File: 523_continuous_subarray_sum/checkSubarraySum.ts
function checkSubarraySum(nums: number[], k: number): boolean {
  let cumulative = 0;

  const map = { "0": -1 };

  for (let i = 0; i < nums.length; i++) {
    cumulative += nums[i];

    if (map[cumulative % k] === undefined) {
      map[cumulative % k] = i;
    }

    if (map[cumulative % k] < i - 1) {
      return true;
    }
  }

  return false;
}

File: 523_continuous_subarray_sum/checkSubarraySum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        map = {
            0: -1
        }
        s = 0

        for i in range(len(nums)):
            s += nums[i]
            if k != 0:
                s %= k
            if s not in map:
                map[s] = i
            elif i - map[s] > 1:
                return True


        return False